sharding/selectionBasedProvider.go

```diff
@@ -52,13 +52,13 @@ func (sbp *SelectionBasedProvider) add(expElList []Validator, index int64) {
 }
 
 // Get will return the consensus group based on the randomness.
-// After a validator is chosen, it is added to the slice so it won't be selected again so next time a new validator
+// After an item is chosen, it is added to the slice so it won't be selected again so next time a new item
 // is needed, the index is recalculated until the validator doesn't exist in that slice
-func (sbp *SelectionBasedProvider) Get(randomness []byte, numValidators int64, expandedEligibleList []Validator) ([]Validator, error) {
+func (sbp *SelectionBasedProvider) Get(randomness []byte, numValidators int64, expandedEligibleList []uint32) ([]uint32, error) {
```

Comment:
because the method uses sbp.size on line 67, I think two consecutive Gets would not work correctly. It will start with the previous sbp.size and it should start from zero.  

`index = newRandomness % uint64(lenExpandedList-sbp.size)`

Reply:
added mutex and cleanup in selection provider to protect in case of concurrency over the same instance
also currently a new instance is created for every selection so concurrent calls through selector are ok

Reply:
ok